Add translation program playbook (#5641)
ethereum/ethereum-org-website: src/content/contributing/translation-program/playbook/index.md

---
title: Translation program playbook
lang: en
description: A collection of tips and important considerations for setting up a translation program
sidebar: true
---

# Translation Program Playbook {#translation-program-playbook}

English is one of the most spoken languages in the world and is by far the world’s most studied language. As English is the most common language used on the internet – especially on social media – and multilingual programming languages are scarce, the majority of content in the blockchain space is natively written in English.

However, as over 6 billion people in the world (more than 75% of the population) do not speak English at all, this presents a massive barrier for entry to Ethereum for the vast majority of the world’s population.

For this reason, an increasing number of projects in the space are looking to get their content translated into different languages and localized for global communities.

Providing multilingual content is a simple and effective way of growing your global community, providing education to non-English speakers, making sure your content and communications reach a wider audience, and onboarding more people to the space.

> This guide aims to address the common challenges and misconceptions about content localization, as well as provide a step-by-step guide to managing content, the translation and review process, quality assurance, translator outreach, and other key aspects of the localization process.

Reviewer: ```suggestion
This guide aims to address the common challenges and misconceptions about content localization. It provides a step-by-step guide to managing content, the translation and review process, quality assurance, translator outreach, and other vital aspects of the localization process.
```

## Content Management {#content-management}

Translation content management refers to the process of automating the translation workflow, which removes the need for repetitive manual work, improves efficiency and quality, allows for better control, and enables collaboration.

There are many different approaches to content management in the localization process, depending on the content and your needs.

The fundamental way of managing content is to create bilingual files, containing the source and target text. This is rarely used in translation, since it offers no significant advantages, apart from simplicity.

Translation agencies usually approach translation management by using translation management software or localization tools, which provide project management capabilities and allow for much greater control over the files, content, and linguists.

Read more about content management:

[Trados on what is translation management](https://www.trados.com/solutions/translation-management/)

[Phrase on multilingual content management](https://phrase.com/blog/posts/multilingual-content-management/)

### Translation Management Software {#translation-management-software}

> There are a number of translation management systems and localization tools, and the choice of software depends largely on your needs.
> 
> While some projects decide against the use of translation management systems and prefer to handle translations manually – either directly in bilingual files or on hosting services, such as GitHub – this greatly reduces control, productivity, quality, scalability, and collaboration capabilities. Such an approach might be most useful for small-scale or one-off translation projects.

Reviewer: ```suggestion
There are many translation management systems and localization tools, and the choice of software depends mainly on your needs.

While some projects decide against using translation management systems and prefer to handle translations manually – either directly in bilingual files or on hosting services, such as GitHub – this dramatically reduces control, productivity, quality, scalability, and collaboration capabilities. Such an approach might be most beneficial for small-scale or one-off translation projects.
```

and many others...

Read more about translation management software:

[Wikipedia definition of translation management systems](https://en.wikipedia.org/wiki/Translation_management_system)

[Phrase on 7 things every translation management software should have](https://phrase.com/blog/posts/7-things-every-translation-management-software-should-have/)

[MemoQ on what is a translation management system](https://www.memoq.com/tools/what-is-a-translation-management-system)

[Gengo’s list of 16 best translation management systems](https://gengo.com/translator-product-updates/16-best-translation-management-systems/)

## Workflow {#workflow}

In the translation space, translation workflow can mean a couple of different things, both somewhat interrelated, and important considerations for your project.

We will explore both of them below.

**Meaning 1**

This is probably the most common way of thinking about translation workflows and something that usually comes to mind when hearing the word workflow.

In its essence, it is the ‘flow of work’ from starting to think about translations to using the translated content in your product.

An example workflow in this case would be:

> 1. **Preparing the files for translation** – It sounds simple, however, there are a couple of important things you need to consider. At this step, you should have a clear plan on how the entire process should work. 
> - *Which file types will you be using? What format do you want to receive your translated files in?*
>     - If your content is available in DOCX or MD format, the approach will be much more straightforward than if you are translating a PDF version of your Whitepaper or other documents.
> - *Which localization tools support this file type? Can the file be translated in a way that retains the original formatting?*
>     - Not all file types support direct localization (e.g., PDF files, image files) and not all localization tools support all file types.
> - *Who will be translating the content? Will you be ordering professional translations, or relying on volunteers?*
>     - This affects a number of other decisions you need to make. For example, professional translators are more comfortable working with advanced localization tools than volunteers.
> - *What are your expectations for the linguists? If you are using a language service provider, what do they expect from you?*
>     - This is the step to make sure your goals, expectations, and timelines are aligned.
> - *Is all the content for translation equally important? Should some content be translated first?*
>     - There are a number of ways to prioritize certain content, which should be translated and implemented first. If you have a lot of content for translation, you can use version control to make sure the translators are aware of which content should be prioritized.
>     
> 2. **Sharing the files for translation** – This step also requires some long-term thinking, and is not as straightforward as sending the source files to a language service provider.
> - *Who will be translating the content? How many people will be involved in this process?*
>     - If you are planning on using a localization tool, this step is simplified, since you can upload the source files to the tool directly. This is also true if the translation process will take place on the hosting service, since the source files don’t need to be exported anywhere.
> - *Will the source files be handled manually, or can this process be automated?*
>     - Most localization tools allow for some type of integration or automation of the file management process. On the other hand, if you are working with individual translators and not using a localization tool, manually sending source files to hundreds or thousands of translators is not a scalable process.
> - *Which tools will be used for the localization?*
>     - The answer to this question will determine how you approach everything else. Selecting the proper tool can help you automate content management, managing the Translation Memory and Glossary, managing translators, keeping track of the translation/review progress, etc., so take some time and do some research on which tool you want to use. If you are not planning on using a localization tool, all of the above will need to be done manually.
> - *How long will the translation process take? How much will it cost?*
>     - At this point, you should be ready to share the source files with the language service provider or pool of translators. The language service provider can help you analyze the word count and provide a quote, including the rates and timeline for the translation process.
> - *Are you planning on making changes/updating the source content during this process?*
>     - If your content is dynamic and changes often, any changes or updates can disrupt the translation progress. Using a Translation Memory can help mitigate this significantly, though it is still important to think about how the process will work and how you can prevent setting back the progress the translators are making.
>         
> 3. **Managing the translation process** – Your work isn’t done once the source content is handed off to the language service provider or the translators. To ensure optimal quality of the translations, content creators should be as involved with the translation process as possible.
> - *How are you planning on communicating with the translators?*
>     - If you are planning on using a localization tool, the communication can take place directly in the tool. Setting up an alternative communication channel with the translators is also recommended, since they might be less hesitant to reach out and messaging tools allow for more free-flowing communication.
> - *How to handle questions from translators? Who should be answering these questions?*
>     - Translators (both professional and non-professional) will often reach out with questions and requests for clarification or additional context, as well as feedback and ideas for improvements. Replying to these inquiries can often lead to better engagement and quality of translated content. It is also valuable to provide them with as many resources as possible (e.g., guides, tips, terminology guidelines, FAQs, etc.).
> - *How to handle the review process? Do you want to outsource it, or do you have the capacity to perform reviews internally?*
>     - While not always necessary, reviews are an integral part of an optimal translation process. Usually, it is easiest to outsource the review process to professional reviewers. However, if you have a large international team, the reviews or QA can also be handled internally.
>     
> 4. **Implementing the translated content** – The last part of the workflow, though still important to consider ahead of time.
> - *Will all the translations be completed at the same time?*
>     - If not, you should think about which translations should be prioritized, how to keep track of the translations in progress, and how the implementation is handled while the translations are done.
> - *How will the translated content be delivered to you? What format will it be in?*
>     - This is an important consideration, regardless of which approach you use. Localization tools allow you to maintain control over the target file format and export process, and usually support automation, e.g., by enabling integration with the hosting service.
> - *How will you be implementing the translations in your project?*
>     - In some cases, this could be as simple as uploading the translated file or adding it to your docs. However, with more complex projects, like website or app translations, you should make sure the code supports internationalization, and establish how the implementation process will be handled ahead of time.
> - *What happens if the formatting is different to the source?*
>     - Similar to the above, if you are translating simple text files, the formatting probably isn’t crucially important. However, with more complex files, like content for a website or application, the formatting and code need to be identical to the source in order to be implemented in your project. If not, the target files will need to be edited, either by the translators or your developers.

Reviewer: ```suggestion
1. **Preparing the files for translation** – It sounds simple; however, you need to consider a couple of important things. At this step, you should have a clear plan on how the entire process should work. 
- *Which file types will you be using? What format do you want to receive your translated files in?*
    - If your content is available in DOCX or MD format, the approach will be much more straightforward than if you are translating a PDF version of your Whitepaper or other documents.
- *Which localization tools support this file type? Can the file be translated in a way that retains the original formatting?*
    - Not all file types support direct localization (e.g., PDF files, image files), and not all localization tools support all file types.
- *Who will be translating the content? Will you be ordering professional translations or relying on volunteers?*
    - This affects a number of other decisions you need to make. For example, professional translators are more comfortable working with advanced localization tools than volunteers.
- *What are your expectations for the linguists? If you are using a language service provider, what do they expect from you?*
    - This is the step to make sure your goals, expectations, and timelines are aligned.
- *Is all the content for translation equally important? Should some content be translated first?*
    - There are some ways to prioritize certain content, which should be translated and implemented first. For example, if you have a lot of content for translation, you can use version control to make sure the translators are aware of which they should prioritize.

2. **Sharing the files for translation** – This step also requires some long-term thinking and is not as straightforward as sending the source files to a language service provider.
- *Who will be translating the content? How many people will be involved in this process?*
    - If you plan to use a localization tool, this step is simplified since you can upload the source files to the tool directly. This is also true if the translation process takes place on the hosting service since the source files don’t need to be exported anywhere.
- *Will the source files be handled manually, or can this process be automated?*
    - Most localization tools allow for some type of integration or automation of the file management process. On the other hand, if you are working with individual translators and not using a localization tool, manually sending source files to hundreds or thousands of translators is not a scalable process.
- *Which tools will be used for the localization?*
    - The answer to this question will determine how you approach everything else. Selecting the proper tool can help you automate content management, managing the Translation Memory and Glossary, managing translators, keeping track of the translation/review progress, etc., so take some time and do some research on which tool you want to use. If you are not planning on using a localization tool, all of the above will need to be done manually.
- *How long will the translation process take? How much will it cost?*
    - At this point, you should be ready to share the source files with the language service provider or pool of translators. The language service provider can help you analyze the word count and provide a quote, including the rates and timeline for the translation process.
- *Are you planning on making changes/updating the source content during this process?*
    - If your content is dynamic and changes often, any changes or updates can disrupt the translation progress. Using a Translation Memory can help mitigate this significantly, though it is still important to think about how the process will work and how you can prevent setting back the progress the translators are making.

3. **Managing the translation process** – Your work isn’t done once the source content is handed off to the language service provider or the translators. To ensure optimal quality of the translations, content creators should be as involved with the translation process as possible.
- *How are you planning on communicating with the translators?*
    - If you are planning on using a localization tool, the communication can take place directly in the tool. Setting up an alternative communication channel with the translators is also recommended since they might be less hesitant to reach out, and messaging tools allow for more free-flowing communication.
- *How to handle questions from translators? Who should be answering these questions?*
    - Translators (both professional and non-professional) will often reach out with questions and requests for clarification or additional context, as well as feedback and ideas for improvements. Replying to these inquiries can often lead to better engagement and quality of translated content. It is also valuable to provide them with as many resources as possible (e.g., guides, tips, terminology guidelines, FAQs, etc.).
- *How to handle the review process? Do you want to outsource it, or do you have the capacity to perform reviews internally?*
    - While not always necessary, reviews are an integral part of an optimal translation process. Usually, it is easiest to outsource the review process to professional reviewers. However, if you have a large international team, the reviews or QA can also be handled internally.

4. **Implementing the translated content** – The last part of the workflow, though still important to consider ahead of time.
- *Will all the translations be completed at the same time?*
    - If not, you should think about which translations should be prioritized, how to keep track of the translations in progress, and how the implementation is handled while the translations are done.
- *How will the translated content be delivered to you? What format will it be in?*
    - This is an important consideration, regardless of which approach you use. Localization tools allow you to maintain control over the target file format and export process and usually support automation, e.g., by enabling integration with the hosting service.
- *How will you be implementing the translations in your project?*
    - In some cases, this could be as simple as uploading the translated file or adding it to your docs. However, with more complex projects, like website or app translations, you should make sure the code supports internationalization and establish how the implementation process will be handled ahead of time.
- *What happens if the formatting is different to the source?*
    - Similar to the above, if you are translating simple text files, the formatting probably isn’t crucially important. However, with more complex files, like content for a website or application, the formatting and code need to be identical to the source in order to be implemented in your project. If not, the target files will need to be edited, either by the translators or your developers.
```

There are a couple of steps to follow when starting to think about terminology management:

- Identify key terms that should be included in the termbase.
- Create a glossary of terms and their definitions.
- Translate the terms and add them to the glossary.
- Check and approve the translations.
- Maintain the glossary and update it with new terms, as they become important.

Read more about terminology management:

[Trados on what is terminology management](https://www.trados.com/solutions/terminology-management/translation-101-what-is-terminology-management.html)

[Language Scientific on why terminology management matters](https://www.languagescientific.com/terminology-management-why-it-matters/#:~:text=Terminology%20management%20is%20the%20process,are%20related%20to%20each%20other.)

[Clear Words Translation on what is terminology management and why it matters](http://clearwordstranslations.com/language/en/what-is-terminology-management/)

### Translation Memory and Glossary {#tm-and-glossary}

> The Translation Memory and Glossary are important tools in the translation industry, and something most language service providers rely on.
> 
> Let’s look at what these terms mean and how they are different to each other:
> 
> **Translation memory (TM)** – A database that automatically stores segments or strings, including longer blocks of text, full sentences, paragraphs, and individual terms, as well as their current and previous translations in every language.
> 
> Most localization tools, translation management systems, and computer-assisted translation tools have built-in translation memories, which can usually be exported and used in other similar tools as well.
> 
> The benefits of using a translation memory include faster translations, better translation quality, the ability to retain certain translations when updating or changing source content, and cheaper translation costs for repetitive content.
> 
> Translation memories work based on a percentage match between different segments, and are usually most useful when two segments contain over 50% of the same content. They are also used to automatically translate repetitive segments, which are 100% matches, thus removing the need to ever translate repetitive content more than once.
> 
> Read more about translation memories:
> 
> [Memsource on translation memories](https://www.memsource.com/translation-memory/)
> 
> [Smartling on what is a translation memory](https://www.smartling.com/resources/101/what-is-translation-memory/)
> 
> **Glossary –** A list of important or sensitive terms, their definitions, functions, and established translations. The main difference between a glossary and a translation memory is that a glossary is not created automatically, and that it does not contain translations of full sentences.
> 
> Most localization tools, translation management systems, and computer-assisted translation tools have built-in glossaries that you can maintain to ensure they contain terminology, important for your project. Similar to the TM, the glossary can usually be exported and used in other localization tools.
> 
> Before starting your translation project, it is highly recommended to take some time and create a glossary for your translators and reviewers. Using a glossary ensures that important terms are translated correctly, provides translators with much needed context, and guarantees consistency in translations.
> 
> While glossaries most often contain established translations in the target languages, they are also useful without this. Even without established translations, a glossary can contain definitions of technical terms, highlight terms that should not be translated, and inform translators whether a certain term is used as a noun, verb, proper noun, or any other part of speech.

Reviewer: ```suggestion
The Translation Memory and Glossary are important tools in the translation industry and something most language service providers rely on.

Let’s look at what these terms mean and how they are different to each other:

**Translation memory (TM)** – A database that automatically stores segments or strings, including longer blocks of text, complete sentences, paragraphs, and individual terms, as well as their current and previous translations in every language.

Most localization tools, translation management systems, and computer-assisted translation tools have built-in translation memories, which can usually be exported and used in other similar tools as well.

The benefits of using a translation memory include faster translations, better translation quality, the ability to retain certain translations when updating or changing source content, and cheaper translation costs for repetitive content.

Translation memories work based on a percentage match between different segments and are usually most useful when two segments contain over 50% of the same content. They are also used to automatically translate repetitive segments, which are 100% matches, thus removing the need to ever translate repetitive content more than once.

Read more about translation memories:

[Memsource on translation memories](https://www.memsource.com/translation-memory/)

[Smartling on what is a translation memory](https://www.smartling.com/resources/101/what-is-translation-memory/)

**Glossary –** A list of important or sensitive terms, their definitions, functions, and established translations. The main difference between a glossary and a translation memory is that a glossary is not created automatically, and that it does not contain translations of full sentences.

Most localization tools, translation management systems, and computer-assisted translation tools have built-in glossaries that you can maintain to ensure they contain terminology important for your project. Like the TM, the glossary can usually be exported and used in other localization tools.

Before starting your translation project, it is highly recommended to take some time and create a glossary for your translators and reviewers. Using a glossary ensures that important terms are translated correctly, provides translators with much-needed context, and guarantees consistency in translations.

While glossaries most often contain established translations in the target languages, they are also useful without this. Even without established translations, a glossary can have definitions of technical terms, highlight terms that should not get translated, and inform translators whether a specific term is used as a noun, verb, proper noun, or any other part of speech.
```
